main.privateendpoint.tf

```diff
@@ -2,7 +2,7 @@
 resource "azurerm_private_endpoint" "this" {
   for_each                      = var.private_endpoints
   name                          = each.value.name != null ? each.value.name : "pe-${var.name}"
-  location                      = each.value.location != null ? each.value.location : var.location
+  location                      = each.value.location != null ? each.value.location : coalesce(var.location, data.azurerm_resource_group.parent[0].location)
```

Comment:
```suggestion
  location                      = coalesce(each.value.location, var.location, data.azurerm_resource_group.parent[0].location)
```

Reply:
I think we can simplify this - wdyt?

Reply:
yeh much better - should have spotted that!